bench/src/simulated_network.rs

```diff
@@ -0,0 +1,250 @@
+use std::collections::VecDeque;
+use std::io::{Error, IoSliceMut};
+use std::net::{IpAddr, Ipv6Addr, SocketAddr};
+use std::sync::{Arc, Mutex};
+use std::task::{Poll, Waker};
+use std::time::{Duration, Instant};
+
+use quinn::udp::{RecvMeta, Transmit, UdpState};
+use quinn::AsyncUdpSocket;
+
+use queue::InboundQueue;
+
+#[derive(Debug)]
+pub struct InMemorySocketHandle {
+    pub network: Arc<InMemoryNetwork>,
+    pub addr: SocketAddr,
+}
+
+impl AsyncUdpSocket for InMemorySocketHandle {
```

Comment:
Should this also implement the recently-introduced `fn may_fragment(&self) -> bool { false }`?